### ISSUE TITLE
Textdraw being displayed out of order

### ISSUE THREAD

**Issue by T3L4** [BenBurak/Textdraw-Streamer#36]
I am using the latest version of the include on a Linux Ubuntu-22.04-x86_64 and this problem doesn't always happen.
In short, when a textdraw is shown, parts of it are out of order. All of them are there, but what should be in front sometimes is behind and gets hidden. I know that those hidden ones are still being shown because of some reports from players on my server who, when reporting the bug (because they have modifications that alter the transparency of the textdraws), can still see them, but they are behind. I don't know if I made myself clear, but some GTA graphic mods change the transparency and appearance of some textdraws, and that's why I conclude that all of them are shown, just out of order. So, the button to buy a vehicle at the dealership, which should be in front of all the others, sometimes when the player accesses the dealership, this button is hidden (behind some other textdraw, not being shown).
I notice that the 'older' textdraws, that is, those that are higher up in the gamemode, do not have this problem. An example of this is our inventory, which never has this bug and always opens perfectly.
Note: This bug happens sometimes with many or very few players on the server... And sometimes it is necessary to open the textdraw 1, 2, 3, 4 times for it to work perfectly.

Any idea how to fix it?

**Comment by nexquery** [comment]
This is the first time I have heard of this problem. 
There seems to be a problem for PreviewModels as well.
I will probably rewrite the plugin in a different way to solve the issues.

**Comment by T3L4** [comment]
Yes, I also have a big problem with PreviewModels. Every time it is necessary to hide to change a model that is already being shown and this impacts the entire operation.

Thank you for responding, I will wait for new updates.

**Comment by GabrielLucioC** [comment]
Any news?
I'm also having the same problem...
From what I understand, if I try to display it on the screen when it's already at the limit, the layer problem ends up occurring, the new layers start to fall behind the old ones.

**Comment by Tela7** [comment]
> Any news? I'm also having the same problem... From what I understand, if I try to display it on the screen when it's already at the limit, the layer problem ends up occurring, the new layers start to fall behind the old ones.

Until it is fixed, I was forced to change all the textdraws that had this issue to TextDrawPlayer. I no longer have this problem.

**Comment by rickzn2** [comment]
I have a similar problem, my textdraw are having problems mainly in previews, colors, and sometimes when I enter my server there is a black screen, something that does not happen with the native system, the previews of my inventory appear in gradient, it does not update many times, and I still haven't found a solution.

**Comment by nexquery** [comment]
> I have a similar problem, my textdraw are having problems mainly in previews, colors, and sometimes when I enter my server there is a black screen, something that does not happen with the native system, the previews of my inventory appear in gradient, it does not update many times, and I still haven't found a solution.

Probably a plugin bug, I will release a new version soon.

**Comment by rickzn2** [comment]
I noticed that Kristo Isbergs version from 6 years ago updates the preview without bugs, but the colors are still buggy for both the preview and everything else. They present many more problems when they need to be updated.

I would be very happy if you could solve the problem as I had to stop developing my server due to these problems.

**Comment by nexquery** [comment]
I will go over all the issues again. Probably all the issues will be fixed in a few weeks.

**Comment by nexquery** [comment]
Try the new version and check if the issue persists.
